app/src/parser.c

```diff
@@ -27,13 +29,22 @@ void __assert_fail(const char * assertion, const char * file, unsigned int line,
 }
 #endif
 
+parser_tx_t parser_state;
+
 parser_error_t parser_parse(parser_context_t *ctx, const uint8_t *data, size_t dataLen) {
-    CHECK_PARSER_ERR(parser_init(ctx, data, dataLen))
-    return _read(ctx, &parser_tx_obj);
+    parser_state.state = NULL;
+    parser_state.len = 0;
+    // Just in case
+    //zb_deallocate();
+    CHECK_PARSER_ERR(_parser_init(ctx, data, dataLen, &parser_state.len))
+    if ( parser_state.len> 0 && zb_allocate(parser_state.len) == zb_no_error) {
```

Comment:
is it possible that parser_state.len is not %4==0 ?
maybe `_parser_init` should pad and overestimate a bit to ensure that we cover the requirements but still len%4=0 to avoid an error due to misalignment

Reply:
yes, we can check it before trying to allocate, and also add a procedure on the Rust site to also check this and approximate the required length to the nearest multiple of 4.

Reply:
the check happend in zb_allocate.. but it would be good to be sure that _parser_init always returns something that is >= to what we need and a multiple of 4

Reply:
yes, done